sycl/include/CL/sycl/INTEL/esimd/detail/esimd_intrin.hpp

```diff
@@ -64,11 +64,11 @@
 //
 template <typename T, int N, int M, int VStride, int Width, int Stride,
           int ParentWidth = 0>
-SYCL_EXTERNAL __SIGD::vector_type_t<T, M>
+SYCL_EXTERNAL SYCL_ESIMD_FUNCTION __SIGD::vector_type_t<T, M>
```

Comment:
I think the suggested way to add function attributes is after the last ')' - @erichkeane, is it correct?

Reply:
No, that adds the attribute to the function *type* and not the function *declaration*.

Reply:
What Aaron said!  For an attribute to apply to the function declaration it needs to go on the left.  See SYCL_EXTERNAL, which is a function declaration attribute.

Reply:
I see, thanks. We'll need to refactor examples and tests.
What would be the right place for the attribute to apply to a lambda?
`single_task<class test>([](item<1> i) {});` ?

Reply:
In a lambda, the only place you can currently put an attribute is in the type position (this was resolved for C++23 but SYCL can't require use of that yet, for obvious reasons). So:
```
single_task<class test>([](item<1> i) [[attr]] {});
```
is the only place for it to go on a lambda for the moment.